core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -48,21 +48,21 @@ private[spark] case class ResourceRequest(
     discoveryScript: Option[String],
     vendor: Option[String])
 
-private[spark] case class TaskResourceRequirement(resourceName: String, amount: Int)
+private[spark] case class ResourceRequirement(resourceName: String, amount: Int)
 
 /**
  * Case class representing allocated resource addresses for a specific resource.
  * Cluster manager uses the JSON serialization of this case class to pass allocated resource info to
  * driver and executors. See the ``--resourcesFile`` option there.
  */
-private[spark] case class ResourceAllocation(id: ResourceID, addresses: Seq[String]) {
+@Evolving
+case class ResourceAllocation(id: ResourceID, addresses: Seq[String]) {
```

Comment:
Sorry I just realized we made this public but REsourceID is private still.  lets make this private again and just put the format in the docs like you had it before.  Again sorry for switching on you here. If we end up thinking users will find this useful then we can open it up to be public later.

Reply:
never mind.